src/sources/mainwindow.cpp

```diff
@@ -13,3 +16,46 @@ MainWindow::~MainWindow()
 {
     delete ui;
 }
+
+void MainWindow::on_actionLoad_Reference_Image_triggered()
+{
+    QString fileName = QFileDialog::getOpenFileName(this,
+                                                     tr("Select Reference Image File"),
+                                                     "",
+                                                     "Images (*.png *.jpg *.tiff);;All Files (*)"
+                                                     );
+    QString status = tr("");
+    if (fileName.length() == 0) status = tr("No File Selected");
+    else status = tr("Selected ") + fileName;
+    statusBar()->showMessage(status);
+}
+
+void MainWindow::on_actionLoad_Current_Image_s_triggered()
+{
+    QStringList files = QFileDialog::getOpenFileNames(this,
+                                                      tr("Select Current Image(s)"),
+                                                      "",
+                                                      "Images (*.png *.jpg *.tiff);;All Files (*)"
+                                                      );
+    QString status = "";
+    switch (files.length()) {
+        case 0: {
+            status = tr("No File Selected");
+        } break;
+
+        case 1: {
+            status = tr("Selected file ") + files.at(0);
+        } break;
+
+        default: {
+            status = status + tr("Selected ") + tr(std::to_string(files.length()).c_str()) + tr(" Files : ");
+            QList<QString>::iterator it = files.begin();
+            status += *it;
+            it++;
+            for (; it != files.end(); it++) {
+                status += ", " + *it;
+            }
+        } break;
+    }
+    statusBar()->showMessage(status);
```

Comment:
Also add timeout for the `showMessage` function
http://doc.qt.io/qt-5/qstatusbar.html#showMessage